File: OneToOneCallSample/app/src/main/java/com/stringee/apptoappcallsample/stringee/service/CallServiceRetryPolicy.java
package com.stringee.apptoappcallsample.stringee.service;

/** Retry policy that forbids background retries and retries for stale sessions. */
final class CallServiceRetryPolicy {
    private static final int MAX_RETRIES = 3;

    private CallServiceRetryPolicy() {
    }

    static boolean shouldRetry(boolean appForeground, boolean sessionOwned, int attempt) {
        return appForeground && sessionOwned && attempt >= 0 && attempt < MAX_RETRIES;
    }
}

File: Kotlin-OneToOneCallSample/app/src/main/java/com/stringee/kotlin_onetoonecallsample/stringee/service/CallServiceRetryPolicy.kt
package com.stringee.kotlin_onetoonecallsample.stringee.service

/** Retry policy that forbids background retries and retries for stale sessions. */
internal object CallServiceRetryPolicy {
    private const val MAX_RETRIES = 3

    @JvmStatic fun shouldRetry(appForeground: Boolean, sessionOwned: Boolean, attempt: Int): Boolean {
        return appForeground && sessionOwned && attempt >= 0 && attempt < MAX_RETRIES
    }
}
